tests/ui/feature-gates/feature-gate-diagnostic-on-const.rs

```diff
@@ -0,0 +1,16 @@
+//! This is an unusual feature gate test, as it doesn't test the feature
+//! gate, but the fact that not adding the feature gate to
+//! the aux crate will cause the diagnostic to not emit the
+//! custom diagnostic message
```

Comment:
Is that strictly necessary? I feel like at least for std, if we annotate with `diagnostic::on_const` we'd want that message to appear even in stable.

Reply:
std has the feature enabled. but in this case the aux crate doesn't have it. So if a user crate doesn't have the feature gate, their uses of `diagnostic::on_const` get ignored

Reply:
Ah! I see. So the feature must be enabled where the annotation is. Gotcha.